src/box/alter.cc

```diff
@@ -1876,18 +1876,44 @@ alter_space_move_indexes(struct alter_space *alter, uint32_t begin,
 					old_def->key_def, alter->pk_def);
 		index_def_update_optionality(new_def, min_field_count);
 		auto guard = make_scoped_guard([=] { index_def_delete(new_def); });
-		if (!index_def_change_requires_rebuild(old_index, new_def))
+		struct index *pk = space_index(old_space, 0);
+		assert(pk != NULL);
+		bool space_is_empty = index_size(pk) == 0;
+		if (!index_def_change_requires_rebuild(old_index, new_def)) {
 			try {
 				(void) new ModifyIndex(alter, old_index, new_def);
 			} catch (Exception *e) {
 				return -1;
 			}
-		else
+		} else if (space_is_empty && !space_is_vinyl(old_space)) {
```

Comment:
I don't like that we explicitly check the space engine here, when we have virtual methods for everything else. Is it always safe to assume that a memtx space is empty if `index_size()` returns 0 in case MVCC is enabled? Maybe, better just disable this case unconditionally, no matter if the space is empty or not?

Reply:
Discussed f2f, decided to disable rebuild for empty spaces as well. Fixed - the branch is removed.